packages/node/src/integrations/http.ts

```diff
@@ -145,21 +147,70 @@ function addRequestBreadcrumb(event: string, url: string, req: http.IncomingMess
 }
 
 /**
- * Function that can combine together a url that'll be used for our breadcrumbs.
+ * Strip the query string and fragment off of a given URL or path (if present)
  *
- * @param options url that should be returned or an object containing it's parts.
- * @returns constructed url
+ * @param urlPath Path including possible query string and/or fragment
+ * @returns Path without query string or fragment
  */
-function extractUrl(options: string | http.ClientRequestArgs): string {
-  if (typeof options === 'string') {
-    return options;
+export function stripUrlPath(urlPath: string): string {
+  if (urlPath.includes('?') || urlPath.includes('#')) {
+    // in theory, the fragment should always come after the query string, but you never know, so make the # a ? so the
+    // split will always happen if either is there
+    return urlPath.replace('#', '?').split('?')[0];
   }
-  const protocol = options.protocol || '';
-  const hostname = options.hostname || options.host || '';
+
+  return urlPath;
+}
+
+/**
+ * Assemble a URL to be used for breadcrumbs and spans.
+ *
+ * @param requestArgs URL string or object containing the component parts
+ * @returns Fully-formed URL
+ */
+export function extractUrl(requestArgs: string | http.ClientRequestArgs): string {
+  if (typeof requestArgs === 'string') {
+    return stripUrlPath(requestArgs);
+  }
+  const protocol = requestArgs.protocol || '';
+  const hostname = requestArgs.hostname || requestArgs.host || '';
   // Don't log standard :80 (http) and :443 (https) ports to reduce the noise
-  const port = !options.port || options.port === 80 || options.port === 443 ? '' : `:${options.port}`;
-  const path = options.path || '/';
-  return `${protocol}//${hostname}${port}${path}`;
+  const port = !requestArgs.port || requestArgs.port === 80 || requestArgs.port === 443 ? '' : `:${requestArgs.port}`;
+  const path = requestArgs.path ? stripUrlPath(requestArgs.path) : '/';
+
+  // internal routes end up with too many slashes
+  return `${protocol}//${hostname}${port}${path}`.replace('///', '/');
```

Comment:
This is a very poor-man's slash normalization.

We'd get this for free if using `new Url(...)`.

Reply:
Can you say more here? I couldn't find any reference to a `Url` class in the MDN docs, and the `URL` class doesn't seem to be able to deal with just a path (even if it starts with three slashes):

![image](https://user-images.githubusercontent.com/14812505/91757221-d2046280-eb82-11ea-9dee-018dffbb7eda.png)

Reply:
When do we even have the case when we have too many slashes? Afaik `ClientRequest` is printing `http` or `https`, without any slashes

Reply:
See my reply below: https://github.com/getsentry/sentry-javascript/pull/2857#discussion_r481159219